### PR TITLE
Add yaml test generator (fuzzer)

### PR DESCRIPTION
**Pull request by boydjohnson**
This PR adds a tool, `bin/gen-scheduler-test`, that can generate a scheduler yaml test according to specifications on the number of addresses used, inputs, outputs, portion of transactions that are invalid, whether inputs and outputs are wildcarded and by how much, etc.

### CODE REVIEW

**Comment by boydjohnson** [comment]
```
usage: gen-yaml-test [-h] [-a ADDRESSES] [--namespace NAMESPACE] [-w WILDCARD]
                     [-t TRANSACTIONS] [-b BATCHES] [-v VALIDITY] [-i INPUTS]
                     [-o OUTPUTS]
                     file

positional arguments:
  file                  The filepath and name relative to
                        validator/tests/test_scheduler/data

optional arguments:
  -h, --help            show this help message and exit
  -a ADDRESSES, --addresses ADDRESSES
                        The number of addresses in total
  --namespace NAMESPACE
                        A namespace, float comma separated pair specifying a
                        namespace to use for some portion of the addresses
  -w WILDCARD, --wildcard WILDCARD
                        A comma separated quad specifying min and max length
                        for inputs then outputs
  -t TRANSACTIONS, --transactions TRANSACTIONS
                        A comma separated pair of numbers for min and max
                        transactions per batch
  -b BATCHES, --batches BATCHES
                        The number of batches to produce
  -v VALIDITY, --validity VALIDITY
                        A number between 0 and 1 specifying the probability of
                        invalid transactions
  -i INPUTS, --inputs INPUTS
                        A comma separated pair specifying the minimum and
                        maximum number of addresses for inputs
  -o OUTPUTS, --outputs OUTPUTS
                        A comma separated pair specifying the minimum and
                        maximum number of addresses for outputs
```

**Comment by boydjohnson** [comment]
Renamed the cli tool to `gen-scheduler-test`.

**Comment by boydjohnson** [comment]
An example of what the cli tool leaves on the top of a file
```
#
#
# Do not edit this file as it is machine generated. It can be reproduced with the following command.
# gen-scheduler-test --seed 9004174202598863464 -a 20 --namespace aaaaaa,.5 -w 70,70,70,70 -t 2,5 -b 20 -v 1.0 -i 2,5 -o 2,5 <filename>
#
#
```

File: validator/tests/test_scheduler/yaml_generator.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+def main(args):
+    opts = parse_args(args=args)
+
+    if opts.seed is None:
+        seed = random.randrange(sys.maxsize)
+    else:
+        seed = int(opts.seed)
+    random.seed(seed)
+
+    sentence = generate_human_readable(opts, seed=seed)
```
Comment (peterschwarz) [inline validator/tests/test_scheduler/yaml_generator.py]: Rename this to "generation_cmd_header"

Comment (boydjohnson) [reply validator/tests/test_scheduler/yaml_generator.py]: Sounds good.

File: validator/tests/test_scheduler/yaml_generator.py [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+def main(args):
+    opts = parse_args(args=args)
+
+    if opts.seed is None:
+        seed = random.randrange(sys.maxsize)
+    else:
+        seed = int(opts.seed)
+    random.seed(seed)
+
+    sentence = generate_cmd_header(opts, seed=seed)
```
Comment (peterschwarz) [inline validator/tests/test_scheduler/yaml_generator.py]: I was actually referring to `sentence`, but I like the new method name too.

`sentence` -> `cmd_header`

Comment (boydjohnson) [reply validator/tests/test_scheduler/yaml_generator.py]: Ah cool. Thanks.

**Pull request merged**